src/bin/generate_doc.rs

```diff
@@ -0,0 +1,11 @@
+fn main() {
+    use stackable_agent::agentconfig::AgentConfig;
+    use std::env;
+    use std::fs;
+    use std::path::PathBuf;
+
+    let target_file = PathBuf::from("documentation/commandline_args.adoc");
+
+    // Unwrap should be fine here, this will currently get called
+    fs::write(&target_file, AgentConfig::get_documentation().unwrap()).unwrap();
```

Comment:
It is not necessary to create a `PathBuf` from the string reference. The references can be converted automatically.

If something goes wrong then `unwrap` does not provide enough information to fix the problem. Perhaps it is better to use `expect` or `unwrap_or_else`:

```suggestion
    let target_file = "documentation/commandline_args.adoc";

    fs::write(target_file, AgentConfig::get_documentation()).unwrap_or_else(|err| {
        panic!(
            "Could not write documentation to [{}]: {}.",
            target_file, err
        )
    });
```

(If `get_documentation` continues to return a `Result` instead of a `String` then `unwrap` or something else must be added again.)